linkerd/core/src/main/scala/io/buoyant/linkerd/RequestActiveTracer.scala

```diff
@@ -0,0 +1,263 @@
+package io.buoyant.linkerd
+
+import com.twitter.finagle._
+import com.twitter.finagle.client.Transporter.EndpointAddr
+import com.twitter.finagle.http._
+import com.twitter.finagle.http.Fields.MaxForwards
+import com.twitter.finagle.http.Status
+import com.twitter.finagle.http.util.StringUtil
+import com.twitter.finagle.naming.buoyant.DstBindingFactory
+import com.twitter.util._
+import io.buoyant.namer.DelegateTree._
+import io.buoyant.namer.{DelegateTree, Delegator}
+import io.buoyant.router.RouterLabel
+import io.buoyant.router.RoutingFactory
+import io.buoyant.router.RoutingFactory.BaseDtab
+import io.buoyant.router.context.{DstBoundCtx, DstPathCtx}
+
+/**
+ * Intercepts Http TRACE requests with a Max-Forwards and l5d-add-context header.
+ * Returns identification, delegation, and service address information
+ * for a given router as well as any other TRACE responses received from downstream
+ * services.
+ *
+ * A router that receives a tracer request with an l5d-max-depth
+ * greater than 0 decrements the l5d-max-depth header by 1 and
+ * forwards the tracer request to a downstream service that may perform
+ * additional processing of the request.
+ *
+ * @param endpoint the endpoint address of a downstream service identified by a router
+ * @param namers namers used to evaluate a service name
+ * @param dtab base dtab used by the router
+ * @param routerLabel name of router
+ */
+class RequestActiveTracer(
+  endpoint: EndpointAddr,
+  namers: DstBindingFactory.Namer,
+  dtab: BaseDtab,
+  routerLabel: String
+) extends SimpleFilter[Request, Response] {
+
+  private[this] case class PathNode(
```

Comment:
this represents one step in the delegation, rigth?  should this be called DelegationNode or something similar?